grind/bin/grind

```diff
@@ -439,11 +441,17 @@ class TestRunner:
         cmd = "%s batcharchive --dump-json=%s --"
         hashes_file = os.path.join(self.output_dir, "hashes.json")
         cmd = cmd % (self.config.isolate_path, hashes_file)
-        logger.debug("Invoking %s", cmd)
-        p = subprocess.Popen(shlex.split(cmd) + i.isolated_files, env=isolate_env)
-        p.wait()
-        if p.returncode != 0:
-            raise Exception("isolate batcharchive failed")
+        for retry in range(0, 3):
+            # retry a couple of times to overcome connection timeouts.
+            logger.debug("Invoking %s", cmd)
+            p = subprocess.Popen(shlex.split(cmd) + i.isolated_files, env=isolate_env)
+            p.wait()
+            if p.returncode != 0:
+                if retry == 2:
```

Comment:
would prefer we count down to 0 (e.g. num_retries_remaining), this way we don't have to update here and the range() if we want to change the # of retries.